src/renderer/viz/expressions/ramp.js

```diff
@@ -135,6 +143,262 @@ export default class Ramp extends BaseExpression {
+     *      defaultOthers: 'Other Vehicles'
+     *   });
+     *
+     *   // legend = [
+     *   //   { name: 'Bicycle', values: [{ r: 95, g: 70, b: 144, a: 1 }] },
+     *   //   { name: 'Car', values: [{ r: 29, g: 105, b: 150, a: 1 }] }
+     *   //   { name: 'Bus', values: [{ r: 56, g: 166, b: 165, a: 1 }] }
+     *   //   { name: 'Other Vehicles', values: [{ r: 15, g: 133, b: 84, a: 1 }] }
+     *   // ]
+     * });
+     *
+     * @example <caption>Get 4 samples for a linear color ramp</caption>
+     * const s = carto.expressions;
+     * const viz = new carto.Viz({
+     *   color: s.ramp(s.linear(s.prop('numeric'), 1, 100), s.palettes.PRISM)
+     * });
+     *
+     * layer.on('loaded', () => {
+     *   const legend = layer.getViz().color.getLegend({
+     *       samples: 4
+     *   });
+     *
+     *   // legend = [
+     *   //   { name: 'numeric', values: [// rgba color, [0, 25 ]] },
+     *   //   { name: 'numeric', values: [// rgba color, [25, 50 ]] }
+     *   //   { name: 'numeric', values: [// rgba color, [50, 75 ]] }
+     *   //   { name: 'numeric', values: [// rgba color, [75, 100 ]] }
+     *   // ]
+     * });
+     *
+     * @example <caption>Get 4 samples for a linear color ramp (String)</caption>
+     * const viz = new carto.Viz(`
+     *   color: ramp(linear($numeric, 1, 100), PRISM)
+     * `);
+     *
+     * layer.on('loaded', () => {
+     *   const legend = layer.getViz().color.getLegend({
+     *       samples: 4
+     *   });
+     *
+     *   // legend = [
+     *   //   { name: 'numeric', values: [// rgba color, [0, 25 ]] },
+     *   //   { name: 'numeric', values: [// rgba color, [25, 50 ]] }
+     *   //   { name: 'numeric', values: [// rgba color, [50, 75 ]] }
+     *   //   { name: 'numeric', values: [// rgba color, [75, 100 ]] }
+     *   // ]
+     * });
+     * @memberof carto.expressions.Ramp
+     * @name getLegend
+     * @instance
+     * @api
+     */
+    getLegend (options) {
+        const config = Object.assign(DEFAULT_OPTIONS, options);
+
+        if (config.samples > MAX_SAMPLES) {
+            throw new Error(`The maximum number of samples for a legend is ${MAX_SAMPLES}`);
+        }
+
+        if (this.input.isA(Linear)) {
+            return this._getLegendLinear(this.input, config);
+        }
+
+        if (this.input.type === inputTypes.CATEGORY) {
+            return this.input.list
+                ? this._getLegendList(config)
+                : this._getLeyendCategories(config);
+        }
+    }
+
+    _getLegendLinear (input, config) {
+        const firstChild = input[input.childrenNames[0]];
+        return firstChild.isA(Property)
+            ? this._getLegendLinearProperty(firstChild, config)
+            : this._getLegendLinear(firstChild, config);
+    }
+
+    _getLegendLinearProperty (property, config) {
+        const name = property.name;
+        const feature = {};
+        const breakpoints = _getBreakpoints(this._metadata.properties[name], config.samples);
```

Comment:
Don't you think it would be better (and easier) to get the min/max values from `linear`. Of course, that won't output always the same as this, but if the user is using `linear` with custom `min`/`max` values is probably for a good reason (like having extreme outliers). In that case, using regular intervals will produce a bad result since the outliers will cause to sample much on the clamped region. Also, I think it's more expected to see values ranges from the linear's min to linear's max. On top of that, I think we would be more generic and would reduce the coupling with linear's internals and property's internals.